tests/browsercontext-network-event.spec.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { browserTest as it, expect } from './config/browserTest';
+
+it('BrowserContext.Events.Request', async ({browser, server}) => {
+  const context = await browser.newContext();
+  const page = await context.newPage();
+  const requests = [];
+  context.on('request', request => requests.push(request));
+  await page.goto(server.EMPTY_PAGE);
+  await page.setContent('<a target=_blank rel=noopener href="/one-style.html">yo</a>');
+  const [popup] = await Promise.all([
+    page.waitForEvent('popup'),
+    page.click('a'),
+  ]);
+  await popup.close();
+  const urls = requests.map(r => r.url());
+  expect(requests.length).toBe(3);
```

Comment:
I'd suggest to remove this line. I see this test failing on some bots, and without this line it is easy to see which request is missing, because of the next `expect`.